core/descriptors/src/org/jetbrains/kotlin/builtins/KotlinBuiltIns.java

```diff
@@ -210,6 +210,8 @@ private Primitives(
         public final FqNameUnsafe intRange = rangesFqName("IntRange");
         public final FqNameUnsafe longRange = rangesFqName("LongRange");
 
+        public final FqNameUnsafe stringsKt = textFqName("StringsKt");
```

Comment:
This is a JVM-specific name, so I prefer not to declare it here, but rather store it as a constant at the use site

Reply:
I ended up changing this with function.fqNameWhenAvailable == FqName("kotlin.text.StringsKt.iterator")

However, this is in a common lowering, so when you say that StringsKt is a JVM specific name that makes me wonder if I cannot check it this way for JS and Native backends?

Reply:
You're right. Unfortunately, it won't work for Native. So it seems like you need to actually compare for both cases: a member function declared in class `kotlin.text.StringsKt` for JVM, and a top level function declared in package `kotlin.text` for Native. You can use `IrDeclaration.isTopLevel` extension which takes this into account, and then `IrElement.getPackageFragment` extension to acquire the containing package and check if it's `kotlin.text`. See [`isLateinitIsInitializedPropertyGetter`](https://github.com/JetBrains/kotlin/blob/e9b2e74b15ac4536915fb2b74ba8bf9dc93b80eb/compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/ir/Ir.kt#L241) for usage example.

Reply:
Done, thanks. Now checking that the function is the no value argument top-level method in kotlin.text with the name "iterator". The matcher will then check that the extension receiver is a character sequence and we can apply the optimization.

I'll see if I can test this on kotlin native tomorrow as well to make sure it works.